File: Math/src/main/scala/com/bnd/math/business/rand/RandomDistributionTest.java
package com.bnd.math.business.rand;

import com.bnd.math.domain.rand.BooleanDensityUniformDistribution;

public class RandomDistributionTest {

	public static void main(String[] args) {
		int size = 20;
		int repetitions = 100000;
		int[] counts = new int[size + 1];
		int[] indexOccurences = new int[size];
		RandomDistributionProvider<Boolean> rdp = RandomDistributionProviderFactory.apply(new BooleanDensityUniformDistribution());
		for (int i = 0; i < repetitions; i++) {
			int count = 0;
			int index = 0;
			for (boolean val : rdp.nextList(size)) {
				if (val) {
					count++;
					indexOccurences[index]++;
				}
				index++;
			}
			counts[count]++;
		}
		int i = 0;
		for (int count : counts) {
			System.out.println(i + ":" + count);
			i++;
		}
		System.out.println();
		i = 0;
		for (int occurence : indexOccurences) {
			System.out.println(i + ":" + occurence);
			i++;
		}
	}
}

File: Math/src/main/scala/com/bnd/math/business/rand/RandomDistributionTest.scala
package com.bnd.math.business.rand

import com.bnd.math.domain.rand.BooleanDensityUniformDistribution
import scala.collection.JavaConversions._

object RandomDistributionTest extends App {

  val size = 20
  val repetitions = 100000
  val counts = new Array[Int](size + 1)
  val indexOccurences = new Array[Int](size)
  val rdp = RandomDistributionProviderFactory.apply(new BooleanDensityUniformDistribution)

  for (i <- 0 to repetitions) {
    var count = 0

    rdp.nextList(size).toSeq.zipWithIndex.map { case (value, index) =>
      if (value) {
        count += 1
        indexOccurences(index) += 1
      }
    }

    counts(count) += 1
  }

  counts.zipWithIndex.foreach { case (count, i) =>
    println(i + ":" + count)
  }
  println

  indexOccurences.zipWithIndex.foreach { case (occurence, i) =>
    println(i + ":" + occurence)
  }
}
